javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1131 @@
+    e.foreach(arrayList += _)
+    arrayList
+  }
+
+  def frequency(c: Collection[_], o: AnyRef): Int =
+    c.count(_ === o)
+
+  def disjoint(c1: Collection[_], c2: Collection[_]): Boolean = {
+    if (c1.size < c2.size)
+      !c1.exists(elem => c2.contains(elem))
+    else
+      !c2.exists(elem => c1.contains(elem))
+  }
+
+  def addAll[T](c: Collection[_ >: T], elements: Array[AnyRef]): Boolean =
+    c.addAll(asJavaCollection(elements.asInstanceOf[Array[T]]))
+
+  def newSetFromMap[E](map: Map[E, Boolean]): Set[E] = {
+    if (map.nonEmpty)
+      throw new IllegalArgumentException
+
+    new WrappedSet[E, Set[E]] {
+      override protected val inner: Set[E] =
+        map.keySet
+
+      override def add(e: E): Boolean =
+        map.put(e, true) === null
+
+      override def addAll(c: Collection[_ <: E]): Boolean =
+        c.foldLeft(false)((prev, elem) => map.put(elem, true) === null || prev)
+    }
+  }
+
+  @inline
+  private def modulo(a: Int, b: Int): Int = ((a % b) + b) % b
+
+  @inline
+  private def makeComparator[T <: jl.Comparable[T]]: Comparator[T] = {
+    new Comparator[T] {
+      final def compare(o1: T, o2: T): Int = o1.compareTo(o2)
+    }
+  }
+
+  @inline
+  private implicit def comparatorToOrdering[E](cmp: Comparator[E]): Ordering[E] = {
+    new Ordering[E] {
+      final def compare(x: E, y: E): Int = cmp.compare(x, y)
+    }
+  }
+
+  @inline
+  private def binarySearchCore[E](list: List[E], compareToKey: E => Int): Int = {
+    def notFound(insertionPoint: Int): Int = {
+      -insertionPoint - 1
+    }
+    list match {
+      case _: RandomAccess =>
+        @tailrec
+        def binarySearch(lo: Int, hi: Int): Int = {
+          if (lo < hi) {
+            val mid = lo + (hi - lo) / 2
+            val cmp = compareToKey(list(mid))
+            if (cmp === 0) mid
+            else if (cmp > 0) binarySearch(lo, mid)
+            else
+              binarySearch(mid + 1, hi)
+          } else {
+            notFound(lo)
+          }
+        }
+        binarySearch(0, list.size)
+      case _ =>
+        @tailrec
+        def linearSearch(iter: Iterator[E], index: Int): Int = {
+          if (iter.hasNext) {
+            val cmp = compareToKey(iter.next())
+            if (cmp == 0)
+              index
+            else if (cmp > 0)
+              notFound(index)
+            else
+              linearSearch(iter, index + 1)
+          } else {
+            notFound(list.size)
+          }
+        }
+        linearSearch(list.iterator, 0)
+    }
+  }
+
+  @inline
+  private def indexOfSubListCore(source: List[_], target: List[_], fromStart: Boolean) = {
+    val targetSize = target.size
+    if (targetSize == 0) {
+      0
+    } else {
+      val indices = 0 to (source.size - targetSize)
+      val indicesInOrder = if (fromStart) indices else indices.reverse
+      indicesInOrder.find { i =>
+        target.iterator.zipWithIndex.forall { case (e, j) => e == source(i + j)}
```

Comment:
This could be quadratic in the size of `source`. I would suggest using precisely the formulation of the JavaDoc, and let the List's `equals` method take care of that:
```scala
source.subList(i, i + target.size).equals(target)
```